src/random.cpp

```diff
@@ -129,20 +130,87 @@ void GetRandBytes(unsigned char* buf, int num)
     }
 }
 
+
+/* /fn int getRDSEED( uint64_t *buff, int num)
+ * /brief Use Intel's SP 800-90B & C compliant Hardware implemented
+ *        instruction to add entropy to the pool use to create keys. 
+ *
+ *        For details, see: https://software.intel.com/content/www/us/en/develop/blogs/the-difference-between-rdrand-and-rdseed.html
+ * 
+ *  /param buff Buffer to fill with random bits.
+ *  /param num  Number of 64-bit words to write to buffer.
+ *  /return     Returns 1 if the hardware returned  a valid stream of random numbers, 0 otherwise.
+ *
+ */
+int getRDSEED( uint64_t *buff, uint32_t num){
+    //Initialize state to 0, or failed.
+    uint32_t  cumulativeStatus = 0;
+
+    #ifdef LOWELEVEL_USE_RDSEEED
+        //Check to see if the processor running this code has 
+        //this Instruction. Most Intel/AMD CPUs after 2015
+        //should have this instruction.
+        uint32_t level, eax, ebx, ecx, edx; 
+        
+        //Level where the RDSEED feature is described
+        level = 1;
+    
+        //Extract feature vector
+        __get_cpuid(level, &eax, &ebx, &ecx, &edx);    
+    
+        //Check if RDSEED exists on this processor
+        if( (ecx & bit_RDSEED) == bit_RDSEED ) {
+    
+            //Re-initialize value to 1, or success 
+            cumulativeStatus = 1;
+    
+            for( uint32_t kk = 0; kk < num; kk++){
+                //Get RBG data
+                unsigned int rng_high, rng_low;
```

Comment:
```suggestion
                uint32_t rng_high, rng_low;
```

Reply:
This one is a matter of style, once again. I do like uintX_t better, so I will change this one.

Reply:
@RichDevX I do want to note that on processors where unsigned int has a smaller bit width than 32-bits, this code never executes because the flags of cpuid instruction corresponding to this feature evaluates to zero.